esphome/components/as3935/as3935.cpp

```diff
@@ -216,6 +223,104 @@ uint32_t AS3935Component::get_lightning_energy_() {
   return pure_light;
 }
 
+// REG0x03, bit [7:6], manufacturer default: 0 (16 division ratio).
+// This function returns the current division ratio of the resonance frequency.
+// The antenna resonance frequency should be within 3.5 percent of 500kHz, and
+// so when modifying the resonance frequency with the internal capacitors
+// (tuneCap()) it's important to keep in mind that the displayed frequency on
+// the IRQ pin is divided by this number.
+uint8_t AS3935Component::get_div_ratio() {
+  ESP_LOGV(TAG, "Calling get_div_ratio");
+  uint8_t reg_val = this->read_register(INT_MASK_ANT);
+  reg_val &= ~DIV_MASK;
+  reg_val >>= 6;  // Front of the line.
+
+  if (reg_val == 0)
+    return 16;
+  else if (reg_val == 1)
+    return 32;
+  else if (reg_val == 2)
+    return 64;
+  else if (reg_val == 3)
+    return 128;
+  ESP_LOGW(TAG, "Unknown response received for div_ratio");
+  return 0;
+}
+
+uint8_t AS3935Component::get_tune_cap() {
+  ESP_LOGV(TAG, "Calling get_tune_cap");
+  uint8_t reg_val = this->read_register(FREQ_DISP_IRQ);
+  return ((reg_val &= ~CAP_MASK) * 8);  // Multiplied by 8pF
+}
+
+// REG0x08, bits [5,6,7], manufacturer default: 0.
+// This will send the frequency of the oscillators to the IRQ pin.
+//  _osc 1, bit[5] = TRCO - System RCO at 32.768kHz
+//  _osc 2, bit[6] = SRCO - Timer RCO Oscillators 1.1MHz
+//  _osc 3, bit[7] = LCO - Frequency of the Antenna
+void AS3935Component::display_oscillator(bool _state, uint8_t osc) {
+  if ((osc < 1) || (osc > 3))
+    return;
+
+  if (_state) {
+    if (osc == 1)
+      this->write_register(FREQ_DISP_IRQ, OSC_MASK, 1, 5);
+    if (osc == 2)
+      this->write_register(FREQ_DISP_IRQ, OSC_MASK, 1, 6);
+    if (osc == 3)
+      this->write_register(FREQ_DISP_IRQ, OSC_MASK, 1, 7);
+  }
+
+  if (!_state) {
+    if (osc == 1)
+      this->write_register(FREQ_DISP_IRQ, OSC_MASK, 0, 5);  // Demonstrative
+    if (osc == 2)
+      this->write_register(FREQ_DISP_IRQ, OSC_MASK, 0, 6);
+    if (osc == 3)
+      this->write_register(FREQ_DISP_IRQ, OSC_MASK, 0, 7);
+  }
```

Comment:
```suggestion
  this->write_register(FREQ_DISP_IRQ, OSC_MASK, _state, 4 + osc);
```

Given that this is always called with constant arguments, the check on `osc` isn't necessary and this could probably be inlined in the callers.